activejob/lib/active_job/core.rb

```diff
@@ -1,6 +1,13 @@
 # frozen_string_literal: true
 
 module ActiveJob
+  # Raised during job payload deserialization when it references an uninitialized job class.
+  class UninitializedJobClassError < NameError
```

Comment:
```suggestion
  class UnknownJobClassError < NameError
```

Sounds like a better name?